### ISSUE TITLE
Allow to reference in a Package to an imgpkgBundle present in a private repo referencing images outside that private repo

### ISSUE THREAD

**Issue by fjagugar** [carvel-dev/kapp-controller#197]
**Describe the problem/challenge you have**

I have a Package that points to an imgpkgBundle which is in a private registry. Currently, using the `secretRef` attribute, I'm able to point to the credentials placed in the cluster to pull that imgpkgBundle. When fetching this imgpkgBundle, kapp-controller seems to validate the images referenced inside the images lock file placed inside the bundle. In order to validate these images, kapp-controller seems to use the same credentials defined for fetching the bundle.

This current logic works if the images referenced by the bundle are in the same registry as the bundle itself. However, if the images are present in a different registry (public or private) this validation step fails, as it is not able to authenticate with the provided credentials to the registry where the images are placed.

**Describe the solution you'd like**

On kubernetes, you can define several secrets in a ServiceAccount for pulling the images your deployments will use. If instead of a `secretRef` we are able to provide at the Package a ServiceAccount for pulling the images, we would be able to configure that ServiceAccount with all the secrets needed for pulling all the images needed for deploying the Package. At the end, once the Package is deployed and the kubernetes deployment is created, it will use a certain ServiceAccount for pulling these images when spinning up the pods, so this ServiceAccount with the needed credentials needs to be present in the namespace, so being able to reuse it also during the Package deployment might be a simple way for solving this.

There might be other possible solutions like allowing to specify extra credentials in the Package itself, but the approach of using a ServiceAccount might be the simplest one in terms of UX. 

**Anything else you would like to add:**

Error got when deploying a Package without the required credentials for pulling the images:

```
Error: Syncing directory '0': Syncing directory '.' with imgpkgBundle contents: Imgpkg: exit status 1 (stderr: Error: Rewriting image lock file: Checking image existance: GET https://index.docker.io/v2/bitnami/dokuwiki/manifests/sha256:dafaecc97c6e9ed6c2791e322a6f8c8472edec2195f5770ac4946b8dcb5ffb6d: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:bitnami/dokuwiki Type:repository]]
```

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by fjagugar** [comment]
I've performed extra investigations on this, and I noticed that what is causing this issue is actually imgpkg tool when pulling the bundle. I've been able to reproduce this by executing imgpkg pull command passing a token, so it seems the issue is not related to kapp-controller and there's nothing that can be changed on kapp-controller for solving this.

~~Based on this, I'm going to close the issue, as it doesn't apply. Sorry for the confusion.~~

EDIT: I noticed that imgpkg is actually able to handle credentials for different registries when using a dockerconfig or when [using env variables](https://carvel.dev/imgpkg/docs/latest/auth/#via-environment-variables).

My understanding is that the piece missing is how kapp-controller will get the required authentication secrets and how will it pass them to imgpkg tool, am I right?
